### ISSUE TITLE
Why is only the latter of 'add_roles' or 'remove_roles' processed when done together?

### ISSUE THREAD

**Issue by CRASNY** [Rapptz/discord.py#539]
```
await self.bot.add_roles(user, discord.utils.get(ctx.message.server.roles, name='Guest'))
await self.bot.remove_roles(user, discord.utils.get(ctx.message.server.roles, name='Unverified'))
```

Running this, only the latter role is applied. Running it again, the former role applies. Why can't they work simultaneously?

**Comment by ButterCheezii** [comment]
Duplicate of #56

**Comment by Vexs** [comment]
Easy fix is by using replace_roles and building the list of roles to give them yourself- not too difficult, just make a copy of their roles, append the role you want to give them, and remove the role you want to remove, then take the resulting list and use client.replace_roles

**Comment by Rapptz** [comment]
Fixed in 84f38b1 for rewrite.
